### PR TITLE
PIL-1050 Storyly improvements: Add onPress event handler to Storyly implementation

### PR DESCRIPTION
**Pull request by divyaBacancy**
https://user-images.githubusercontent.com/66996635/121167330-7680a700-c86f-11eb-9e42-f6ca48d74d65.mov

### CODE REVIEW

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just a few comments but overall great stuff! Thanks for the knowledge base webview naming change!

File: src/components/Stories/Stories.js [hunk 1]
```diff
@@ -48,6 +54,24 @@ const Stories = () => {
 
   const logStoryOpen = () => dispatch(logEventAction('STORY_OPEN'));
 
+  const handleEvent = ({ nativeEvent }) => {
+    if (storylyRef.current) {
```
[the remark this answers is not in the document]
Comment (divyaBacancy) [reply src/components/Stories/Stories.js:58]: @IAmKio We have used useRef hook where reference of Storyly instance is stored in current.

[the remark this answers is not in the document]
Comment (IAmKio) [reply src/components/Stories/Stories.js:58]: Cheers!

File: src/components/Stories/Stories.js [hunk 2]
```diff
@@ -48,6 +54,24 @@ const Stories = () => {
 
   const logStoryOpen = () => dispatch(logEventAction('STORY_OPEN'));
 
+  const handleEvent = ({ nativeEvent }) => {
```
[the remark this answers is not in the document]
Comment (divyaBacancy) [reply src/components/Stories/Stories.js]: Done.

File: src/components/Stories/Stories.js [hunk 3]
```diff
@@ -48,6 +54,24 @@ const Stories = () => {
...
+    const mediaTitle = nativeEvent.title;
+    const pathName = mediaURL ? mediaURL.split('://').pop() : '';
+    if (mediaURL && isValidURL(mediaURL)) {
+      navigation.navigate(RoutePath.WEB_VIEW, {
+        title: mediaTitle,
+        url: mediaURL,
+      });
+    } else {
+      Object.keys(RoutePath).includes(pathName) ? navigation.navigate(pathName)
+        : reportOrWarn('Storyly handleEvent error');
```
[the remark this answers is not in the document]
Comment (divyaBacancy) [reply src/components/Stories/Stories.js]: Done.

File: src/utils/validators.js [hunk 4]
```diff
@@ -127,6 +127,11 @@ export function isValidFiatValue(value: string) {
   return regex.test(value);
 }
 
+export function isValidURL(url: string) {
+  const regex = /^(ftp|http|https):\/\/[^ "]+$/;
```
[the remark this answers is not in the document]
Comment (divyaBacancy) [reply src/utils/validators.js]: Done.

**Pull request merged**